mlx/io/gguf.cpp

```diff
@@ -12,6 +15,9 @@ extern "C" {
 
 namespace mlx::core {
 
+// https://github.com/antirez/gguf-tools/blob/main/gguflib.h#L102-L108
+constexpr int gguf_array_header = 12;
```

Comment:
When I first read this I thought it was the constant that represents the array type. Perhaps rename it to `gguf_array_header_size` to make it more explicit?